Sources/AudioKitUI/Visualizations/FFTView.swift

```diff
@@ -77,51 +63,75 @@ public struct FFTView: View {
     private var node: Node
-    private var numberOfBars: Int
-    private var minAmplitude: Double
-    private var maxAmplitude: Double
+    private var barCount: Int
+    private var fftValidBinCount: FFTValidBinCount?
+    private var minAmplitude: Float
+    private var maxAmplitude: Float
+    private let defaultBarCount: Int = 64
+    private let maxBarCount: Int = 128
 
     public init(_ node: Node,
                 linearGradient: LinearGradient = LinearGradient(gradient: Gradient(colors: [.red, .yellow, .green]),
                                                                 startPoint: .top,
                                                                 endPoint: .center),
                 paddingFraction: CGFloat = 0.2,
                 includeCaps: Bool = true,
-                numberOfBars: Int = 50,
-                maxAmplitude: Double = -10.0,
-                minAmplitude: Double = -150.0)
+                validBinCount: FFTValidBinCount? = nil,
+                barCount: Int? = nil,
+                maxAmplitude: Float = -10.0,
+                minAmplitude: Float = -150.0)
     {
         self.node = node
         self.linearGradient = linearGradient
         self.paddingFraction = paddingFraction
         self.includeCaps = includeCaps
-        self.numberOfBars = numberOfBars
         self.maxAmplitude = maxAmplitude
         self.minAmplitude = minAmplitude
+        self.fftValidBinCount = validBinCount
 
         if maxAmplitude < minAmplitude {
             fatalError("Maximum amplitude cannot be less than minimum amplitude")
         }
         if minAmplitude > 0.0 || maxAmplitude > 0.0 {
             fatalError("Amplitude values must be less than zero")
         }
+
+        if let requestedBarCount = barCount {
+            self.barCount = requestedBarCount
+        } else {
+            if let fftBinCount = fftValidBinCount {
+                if Int(fftBinCount.rawValue) > maxBarCount - 1 {
+                    self.barCount = maxBarCount
+                } else {
+                    self.barCount = Int(fftBinCount.rawValue)
+                }
+            } else {
+                self.barCount = defaultBarCount
+            }
+        }
     }
 
     public var body: some View {
         HStack(spacing: 0.0) {
-            ForEach(fft.amplitudes.indices, id: \.self) { number in
-                if let amplitude = fft.amplitudes[number] {
-                    AmplitudeBar(amplitude: amplitude,
+            ForEach(0 ..< barCount) {
+                if $0 < fft.amplitudes.count {
+                    if let amplitude = fft.amplitudes[$0] {
+                        AmplitudeBar(amplitude: amplitude,
+                                     linearGradient: linearGradient,
+                                     paddingFraction: paddingFraction,
+                                     includeCaps: includeCaps)
+                    }
+                } else {
+                    AmplitudeBar(amplitude: 0.0,
```

Comment:
I saw it as a guard against index out of bounds errors that I wasn't anticipating and also it would help developers who didn't understand the difference between bins and bars by seeing visually that there are more bars than bin data returned

My push-back against simplifying to only bins is that I could see someone wanting a low bar count but some higher number of bins. If their view is not screen sized, maybe they need to only display 16 or 32 bars, but want those bars to show the lower fundamental frequencies (thus needing more bins).